assets/js/components/event/StreamArea.tsx

```diff
@@ -6,17 +6,23 @@ import type { Mode, Client } from "../../types/types";
 import { Channel } from "phoenix";
 import useCheckScreenType from "../../utils/useCheckScreenType";
 import "../../../css/event/streamarea.css";
+import { syncAmIPresenter } from "../../utils/modePanelUtils";
+
+import { switchAskingForBeingPresenter } from "../../utils/channelUtils";
+import MobileHlsBar from "./MobileHlsBar";
 
 type StreamAreaProps = {
   client: Client;
   eventChannel: Channel | undefined;
   privateChannel: Channel | undefined;
   mode: Mode;
   setMode: React.Dispatch<React.SetStateAction<Mode>>;
+  eventTitle: string;
 };
 
-const StreamArea = ({ client, eventChannel, privateChannel, mode, setMode }: StreamAreaProps) => {
+const StreamArea = ({ client, eventChannel, privateChannel, mode, setMode, eventTitle }: StreamAreaProps) => {
   const [hlsUrl, setHlsUrl] = useState<string>("");
+  const [amIPresenter, setAmIPresenter] = useState<boolean>(false);
```

Comment:
I agree that it can vastly improve the readability and performance of the code, _but_
The one component that doesn't use `eventChannel`that is `PresenterArea` and it cannot be rerendered every time `eventChannel` is changed. Constant refreshing causes bugs on webrtc since establishing the connection is in the `PresenterArea` component.In short: Rerendering this element causes reconnection to webrtc which then causes errors. 
Nevertheless, I've tried to fix that by putting `presenters` state in the `StreamArea.tsx` but that didn't help since this structure, the `event channel` and `webrtc` are strongly coupled in the `StreamArea` component and its children. It is possible but requires significantly more time that a few minutes. I suggest making an issue on Jira regarding decoupling of the given structures and puting webrtc in the highest component possible.